pyro/distributions/multivariate_studentt.py

```diff
@@ -0,0 +1,95 @@
+        self.loc = loc
+        self.scale_tril = scale_tril
+        self._chi2 = Chi2(self.df)
+        super(MultivariateStudentT, self).__init__(batch_shape, event_shape, validate_args=validate_args)
+
+    @lazy_property
+    def covariance_matrix(self):
+        # NB: this is not covariance of this distribution;
+        # the actual covariance is df / (df - 2) * covariance_matrix
+        return torch.matmul(self.scale_tril, self.scale_tril.transpose(-1, -2))
+
+    @lazy_property
+    def precision_matrix(self):
+        identity = torch.eye(self.loc.size(-1), device=self.loc.device, dtype=self.loc.dtype)
+        scale_inv = identity.triangular_solve(self.scale_tril, upper=False).solution.transpose(-1, -2)
+        return torch.matmul(scale_inv.transpose(-1, -2), scale_inv)
+
+    def expand(self, batch_shape, _instance=None):
+        new = self._get_checked_instance(MultivariateStudentT, _instance)
+        batch_shape = torch.Size(broadcast_shape(self.batch_shape, batch_shape))
+        loc_shape = batch_shape + self.event_shape
+        scale_shape = loc_shape + self.event_shape
+        new.df = self.df.expand(batch_shape)
+        new.loc = self.loc.expand(loc_shape)
+        new.scale_tril = self.scale_tril.expand(scale_shape)
+        if 'covariance_matrix' in self.__dict__:
+            new.covariance_matrix = self.covariance_matrix.expand(scale_shape)
+        if 'precision_matrix' in self.__dict__:
+            new.precision_matrix = self.precision_matrix.expand(scale_shape)
+        new._chi2 = self._chi2.expand(batch_shape)
+        super(MultivariateStudentT, new).__init__(batch_shape, self.event_shape, validate_args=False)
+        new._validate_args = self._validate_args
+        return new
+
+    def rsample(self, sample_shape=torch.Size()):
+        shape = self._extended_shape(sample_shape)
+        X = torch.empty(shape, dtype=self.df.dtype, device=self.df.device).normal_()
+        Z = self._chi2.rsample(sample_shape)
+        Y = X * torch.rsqrt(Z / self.df).unsqueeze(-1)
+        return self.loc + self.scale_tril.matmul(Y.unsqueeze(-1)).squeeze(-1)
+
+    def log_prob(self, value):
+        if self._validate_args:
+            self._validate_sample(value)
+        n = self.loc.size(-1)
+        y = (value - self.loc).unsqueeze(-1).triangular_solve(self.scale_tril, upper=False).solution.squeeze(-1)
+        Z = (self.scale_tril.diagonal(dim1=-2, dim2=-1).log().sum(-1) +
+             0.5 * n * self.df.log() +
+             0.5 * n * math.log(math.pi) +
+             torch.lgamma(0.5 * self.df) -
+             torch.lgamma(0.5 * (self.df + n)))
+        return -0.5 * (self.df + n) * torch.log1p(y.pow(2).sum(-1) / self.df) - Z
+
+    @property
+    def mean(self):
+        m = self.loc.clone()
+        m[self.df <= 1, :] = float('nan')
+        return m
+
+    @property
+    def variance(self):
+        m = self.scale_tril.pow(2).sum(-1) * self.df / (self.df - 2)
+        m[self.df <= 2, :] = float('nan')
```

Comment:
`float('inf')`, right?

Reply:
Yeah, I think it will be inf for 1 < df <=2 like the univariate case. I’ll make a change.